WIP - [Docs] Add page on Deploy

eshurakov opened the pull request:
## Context

This PR adds Deploy page to the Advanced Usage section of the Docs.

eshurakov committed:
Add deployment guide to advanced usage docs

eshurakov committed:
Update deploy.md

bturcotte520 committed:
Enhancing Deploy Docs, removing org constraint

Updated the deployment documentation for Kilo Code, enhancing clarity on deployment processes, prerequisites, and features.

bturcotte520 committed:
Add 'advanced-usage/deploy' to sidebars

bturcotte520 reviewed (commented):
Merged @eshurakov's draft with mine - removed org-only constraint

alexkgold committed:
Fix typo in Kilo Code Deploy description

alexkgold reviewed (changes requested):
Few little things.

bturcotte520 commented on apps/kilocode-docs/docs/advanced-usage/deploy.md:
```diff
@@ -0,0 +1,90 @@
+
+# Deploy
+
+Kilo Deploy lets you ship **Next.js** applications directly from Kilo Code with automatic builds, real-time logs, and continuous deployment from GitHub. It automatically detects your stack, configures your project for hosting, and manages the full deployment lifecycle.
```
Yes! Drafted this last week. Thanks

bturcotte520 committed:
Revise Kilo Deploy documentation and add environment variables section

Updated the Kilo Deploy documentation to improve clarity and added information about environment variables.

alexkgold reviewed (approved):
LGTM
